areas/mysql/src/AzureMcp.MySql/Services/MySqlService.cs

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.MySql.FlexibleServers;
+using AzureMcp.Core.Services.Azure;
+using AzureMcp.Core.Services.Azure.ResourceGroup;
+using AzureMcp.Core.Services.Azure.Tenant;
+using AzureMcp.MySql.Json;
+using Microsoft.Extensions.Logging;
+using MySqlConnector;
+
+namespace AzureMcp.MySql.Services;
+
+public class MySqlService(IResourceGroupService resourceGroupService, ITenantService tenantService, ILogger<MySqlService> logger) : BaseAzureService(tenantService), IMySqlService
+{
+    private readonly IResourceGroupService _resourceGroupService = resourceGroupService ?? throw new ArgumentNullException(nameof(resourceGroupService));
+    private readonly ILogger<MySqlService> _logger = logger;
+    private string? _cachedEntraIdAccessToken;
+    private DateTime _tokenExpiryTime;
+    private readonly object _tokenLock = new object();
+
+    private async Task<string> GetEntraIdAccessTokenAsync()
+    {
+        lock (_tokenLock)
+        {
+            if (_cachedEntraIdAccessToken != null && DateTime.UtcNow < _tokenExpiryTime)
+            {
+                return _cachedEntraIdAccessToken;
+            }
+        }
+
+        var tokenRequestContext = new TokenRequestContext(new[] { "https://ossrdbms-aad.database.windows.net/.default" });
+        var tokenCredential = await GetCredential();
+        var accessToken = await tokenCredential
+            .GetTokenAsync(tokenRequestContext, CancellationToken.None)
+            .ConfigureAwait(false);
+
+        lock (_tokenLock)
+        {
+            _cachedEntraIdAccessToken = accessToken.Token;
+            _tokenExpiryTime = accessToken.ExpiresOn.UtcDateTime.AddSeconds(-60); // Subtract 60 seconds as a buffer.
+        }
+
+        return _cachedEntraIdAccessToken;
+    }
+
+    private static string NormalizeServerName(string server)
+    {
+        if (!server.Contains('.'))
+        {
+            return server + ".mysql.database.azure.com";
+        }
+        return server;
```

Comment:
Is this check safe? What happens if I pass `my.sql.server`?

Reply:
The server name can contain only lowercase letters, numbers, and the hyphen (-) character. It must contain between 3 and 63 characters

https://learn.microsoft.com/en-us/azure/mysql/flexible-server/quickstart-create-server-portal